### ISSUE TITLE
option to use task [meta]title as the display name

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-ui#1072]
#### tldr

Consider adding an option to the tree view to use the `[meta]title` rather than the task name.

#### Explanation

For large `rose-stem` workflows with multidimensional test matrices task names can get rather long.

For example if your test matrix looks like this:

* system
* platform
* option

Then you might end up with a workflow along the lines of this:

```ini
[runtime]
    [[<system>]]
    [[<platform>]]
    [[<option>]]
    [[<system, platform, option>]]
        inherit = <system>, <platform>, <option>
```

With task names like this:

```
systema_platform1_optionx_dosomething
```

Long but manageable, however, I've seen an example with 8+ dimensions! This is just too much for a task name (in fact it's hitting certain system limits).

Because of inheritance these dimensions are already nicely broken down in the tree view:

* SYSTEM_A
  * PLATFORM_1
    * OPTION_X
      * systema_platform1_optionx_dosomething

Providing all of the dimensions in the task name is unnecessary duplication, really it's `SYSTEM_A/PLATFORM_1/OPTION_X/dosomething` all the user wants/needs to see is this:

* SYSTEM_A
  * PLATFORM_1
    * OPTION_X
      * dosomething

The best approach I can think of for handling these issues is to have something like this:

```
[runtime]
    [[dosomething_{{ hash(system, platform, option) }}]]
        [[[meta]]]
            title = Do Something
            system = {{ system }}
            platform = {{ platform }}
            option = {{ option }}
```

Resulting in a short task name like `dosomething_a1g343hk45`, but in order to do this the `[meta]` would need to become more prominent in the GUI. See also #1071.

#### Historical Context

Similar things have been considered in the past. Hillary even had a branch which used the static visualisation colours (a Cylc 7 form of visual metadata) in the Cylc 7 tree view.

**Pull requests welcome!**
